File: test/cases/semantic_valid/sv07_proc_call.pas
program sv07_proc_call;
var
    a, b: integer;

procedure swap(var x, y: integer);
var
    tmp: integer;
begin
    tmp := x;
    x := y;
    y := tmp
end;

begin
    a := 1;
    b := 2;
    swap(a, b)
end.

File: test/cases/semantic_valid/sv07_proc_call.c
#include <stdio.h>
int a;
int b;


void swap(int *x, int *y);

void swap(int *x, int *y) {
    int tmp;

    tmp = (*x);
    (*x) = (*y);
    (*y) = tmp;

}


int main(void) {
    a = 1;
    b = 2;
    swap(&a, &b);

    return 0;
}
